compliance/compliance.go

```diff
@@ -2002,3 +2022,74 @@ func ForwardReferencesDisallowed(c *fluent.GRIBIClient, t testing.TB, _ ...TestO
 			AsResult(),
 		chk.IgnoreOperationID())
 }
+
+// IPv4EntryInvalidPrefix - Replace 203.0.113.1/32 with a syntatically invalid IP address.
+func IPv4EntryInvalidPrefix(c *fluent.GRIBIClient, t testing.TB, _ ...TestOpt) {
+	defer flushServer(c, t)
+
+	ops := []func(){
+		func() {
+			c.Modify().AddEntry(t, fluent.NextHopEntry().WithNetworkInstance(defaultNetworkInstanceName).WithIndex(1).WithIPAddress("192.0.2.3"))
+			c.Modify().AddEntry(t, fluent.NextHopGroupEntry().WithNetworkInstance(defaultNetworkInstanceName).WithID(11).AddNextHop(1, 1))
+			// NB: 203.0.113.1 IPv4Entry should include a CIDR mask
+			c.Modify().AddEntry(t, fluent.IPv4Entry().WithPrefix("203.0.113.1").WithNetworkInstance(defaultNetworkInstanceName).WithNextHopGroup(11))
+		},
+	}
+
+	res := DoModifyOps(c, t, ops, fluent.InstalledInFIB, false)
+
+	chk.HasResult(t, res,
+		fluent.OperationResult().
+			WithIPv4Operation("203.0.113.1").
+			WithOperationType(constants.Add).
+			WithProgrammingResult(fluent.ProgrammingFailed).
+			AsResult(),
+		chk.IgnoreOperationID())
+}
+
+// IPv4EntryMissingNextHopGroup - Missing NextHopGroup for the IPv4Entry 203.0.113.1/32.
+func IPv4EntryMissingNextHopGroup(c *fluent.GRIBIClient, t testing.TB, _ ...TestOpt) {
+	defer flushServer(c, t)
+
+	ops := []func(){
+		func() {
+			c.Modify().AddEntry(t, fluent.NextHopEntry().WithNetworkInstance(defaultNetworkInstanceName).WithIndex(1).WithIPAddress("192.0.2.3"))
+			// NB: NHG 11 has not been defined
+			c.Modify().AddEntry(t, fluent.IPv4Entry().WithPrefix("203.0.113.1/32").WithNetworkInstance(defaultNetworkInstanceName).WithNextHopGroup(11))
+		},
+	}
+
+	res := DoModifyOps(c, t, ops, fluent.InstalledInFIB, false)
+
+	chk.HasResult(t, res,
```

Comment:
Similarly to the above, I'd also check that the entries that you expect to be successfully programmed are (i.e., check that the `203.0.113.1/32` entry being invalid did not affect NHG and NHs being programmed).